o3de/o3de
# feat: replace usage of XmlNoderef with fastxml for SettingsManagerDialog

## What does this PR do?

This just attempts to replace some logic with fastxml. I tried to make some logic more clear by splitting the caller into two functions. This makes the intent of the call more clear less to follow having to know that an empty second parameter means the list of tools comes from settings manager and a non empty list means the user wants to filter against that. second case doesn't need to involve the class itself so it makes sense to move it to a static method. 

## How was this PR tested?
- Edit > Editor Settings > Editor Settings Manager
- export layout to file 
- verify that the settings layout can be both imported/exported.
